girder_worker/app.py

```diff
@@ -129,11 +143,58 @@ def describe(self):
     def call_item_task(self, inputs, outputs={}):
         return self.run.call_item_task(inputs, outputs)
 
+    def _maybe_transform_result(self, idx, result):
+        try:
+            grh = self.request.girder_result_hooks[idx]
+            if hasattr(grh, 'transform') and \
+               hasattr(grh.transform, '__call__'):
+                return grh.transform(result)
+        except IndexError:
+            return result
+
+    def _maybe_transform_argument(self, arg):
+        if hasattr(arg, 'transform') and hasattr(arg.transform, '__call__'):
+            return arg.transform()
+        return arg
+
+    def _maybe_cleanup(self, arg):
+        if hasattr(arg, 'cleanup') and hasattr(arg.cleanup, '__call__'):
+            arg.cleanup()
+
+    def __call__(self, *args, **kwargs):
+        try:
+            _t_args = _walk_obj(args, self._maybe_transform_argument)
+            _t_kwargs = _walk_obj(kwargs, self._maybe_transform_argument)
+
+            results = super(Task, self).__call__(*_t_args, **_t_kwargs)
 
+            if hasattr(self.request, 'girder_result_hooks'):
+                if not isinstance(results, tuple):
+                    results = (results, )
+
+                results = tuple([self._maybe_transform_result(i, r)
+                                 for i, r in enumerate(results)])
+
+            return results
+        finally:
+            _walk_obj(args, self._maybe_cleanup)
+            _walk_obj(kwargs, self._maybe_cleanup)
+
+
+def _maybe_model_repr(obj):
+    try:
+        return obj._repr_model_()
+    except AttributeError:
+        return obj
```

Comment:
We should probably explicitly checked for the existence of ```_repr_model```. Otherwise, an ```AttributeError``` raised in the users ```_repr_model``` implementation it might be hard to track down. This has bitten me before.